mlflow/pyfunc/__init__.py

```diff
@@ -628,6 +633,35 @@ def __repr__(self):
         return yaml.safe_dump({"mlflow.pyfunc.loaded_model": info}, default_flow_style=False)
 
 
+def check_requirements_and_local_installed_mismatch(local_path):
```

Comment:
I'd rename this function to `_warn_requirement_mismatches` to clarify this function emits a warning.